File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int cheapest=INT_MAX;
        int maxprofit=0;
        for(int i=0;i<prices.size();i++){
            if(prices[i]<cheapest){
                cheapest=prices[i];
            }
            maxprofit=max(maxprofit,prices[i]-cheapest);
        }
        return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit=0
        minprice=float('inf')
        
        for i in range(len(prices)):
            minprice=min(minprice,prices[i])
            profit=prices[i]-minprice
            maxprofit=max(profit,maxprofit)
        return maxprofit            
        